### DISCUSSION TITLE
Manual call PutFile from WopiHost

### DISCUSSION THREAD

**Discussion by nautilux2** [petrsvihlik/WopiHost#161] [Q&A]
Hi @petrsvihlik 
I'm using OOS 2016 and your WopiHost. When user(s) type something in an excel file and saves it, then open it again immediately, it's blank because it isn't updated from cache. How can I manual call PutFile from WopiHost to save change when user(s) save, navigate other views ...

I try some of this 

`http://wopi.internal/wopi/files/LlxjNWZhZGEwYy00OTlkLTRlYjYtOTU3Ni1kOWRjOGQzYTM0MWQueGxzeA==/contents?access_token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1laWQiOiIxMjM0NSIsInVuaXF1ZV9uYW1lIjoiQW5vbnltb3VzIiwiZW1haWwiOiJhbm9ueW1vdXNAZG9tYWluLnRsZCIsIlVzZXJQZXJtaXNzaW9ucyI6IlVzZXJDYW5BdHRlbmQsIFVzZXJDYW5QcmVzZW50LCBVc2VyQ2FuUmVuYW1lLCBVc2VyQ2FuV3JpdGUiLCJuYmYiOjE2MTA2MDU5NDgsImV4cCI6MTYxMDYwOTU0OCwiaWF0IjoxNjEwNjA1OTQ4fQ.KbxGcTVOa3yWCGcEHEGv9MRP_NskfViW_vyMTOupRO4&access_token_ttl=0 `

and the wopi host log return status code 200 but the physical file does not change

Thanks

**Comment by petrsvihlik** [comment]
Hi @nautilux2 ,
thanks for submitting the question. Is it in any way related to this StackOverflow question [Autosave file opened in WOPI for editing after an interval](https://stackoverflow.com/questions/66474681/autosave-file-opened-in-wopi-for-editing-after-an-interval?noredirect=1#comment118115576_66474681)?

Does this issue manifest itself only with Excel files? If so, it could be related to this mention in the [documentation](https://wopi.readthedocs.io/en/latest/scenarios/customization.html#checkfileinfo-properties)
>Excel for the web does not guarantee that the latest document content is saved back to the WOPI host before downloading the file.

AFAIK, there is [no way](https://wopi.readthedocs.io/en/latest/faq/disable_autosave.html?highlight=save#can-auto-save-be-disabled-in-wac) to force the WOPI Client (OOS) to call `PutFile` on the WOPI host to flush the unsaved data.

**Comment by nautilux2** [comment]
Hi @petrsvihlik 
It's somehow similar. My case's about user(s) create file, put some text, create table and then they want to transform with an API in backend, after that, backend API returns the transformed file to user. But when user(s) execute too fast, sometime excel file is not sync and still blank.
